sqlglot/dialects/snowflake.py

```diff
@@ -69,6 +69,20 @@ def _unix_to_time(self, expression):
     raise ValueError("Improper scale for timestamp")
 
 
+# https://docs.snowflake.com/en/sql-reference/functions/date_part.html
+# https://docs.snowflake.com/en/sql-reference/functions-date-time.html#label-supported-date-time-parts
+def _parse_date_part(self):
+    this = self._parse_var() or self._parse_type()
+    self._match(TokenType.COMMA)
+    expression = self._parse_bitwise()
+
+    if this.name.upper().startswith("EPOCH"):
```

Comment:
why do you do startswith? shouldn't you check epoch_seconds and milliseconds?

Reply:
I started by checking those (+micro, nano), but I was unsure how to maintain the precision level because we're transforming the expression by using casts.

Reply:
well that just depends on how other formats store time, should probably be fine